docs/Dockerfile.docs

```diff
@@ -27,27 +27,50 @@
 FROM ubuntu:22.04
 
 # various documentation dependencies
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends python3-pip python3-dev git \
-    git-lfs zip doxygen build-essential unzip wget pandoc ssh && \
-    rm -rf /var/lib/apt/lists/*
+RUN apt-get update -q=2 \
+    && apt-get install -y --no-install-recommends \
+        build-essential \
+        doxygen \
+        git \
+        git-lfs \
+        pandoc \
+        python3-dev \
+        python3-pip \
+        ssh \
+        unzip \
+        wget \
+    && rm -rf /var/lib/apt/lists/*
 
 # install protobuf
-ADD https://github.com/google/protobuf/releases/download/v3.6.1/protoc-3.6.1-linux-x86_64.zip ./
-RUN unzip protoc-3.6.1-linux-x86_64.zip -d ./usr/local && \
-  rm protoc-3.6.1-linux-x86_64.zip
+RUN wget https://github.com/google/protobuf/releases/download/v3.6.1/protoc-3.6.1-linux-x86_64.zip -O /tmp/proto.zip \
+    && unzip /tmp/proto.zip -d /usr/local \
+    && rm /tmp/proto.zip
 
 # install pseudomuto/protoc-gen-doc
-RUN wget https://github.com/pseudomuto/protoc-gen-doc/releases/download/v1.3.2/protoc-gen-doc-1.3.2.linux-amd64.go1.12.6.tar.gz && \
-    tar xzf protoc-gen-doc-1.3.2.linux-amd64.go1.12.6.tar.gz && \
-    mv protoc-gen-doc-1.3.2.linux-amd64.go1.12.6/protoc-gen-doc /usr/local/bin/
+RUN wget https://github.com/pseudomuto/protoc-gen-doc/releases/download/v1.3.2/protoc-gen-doc-1.3.2.linux-amd64.go1.12.6.tar.gz -O /tmp/protoc-gen-doc.tar.gz \
+    && tar -xvf /tmp/protoc-gen-doc.tar.gz --strip-components=1 -C /usr/local/bin/ \
+    && rm /tmp/protoc-gen-doc.tar.gz
 
 # install sphinx et al
-RUN pip3 install sphinx==4.5.0 nbclient==0.5.13 \
-    docutils==0.16 ablog==0.10.33.post1 myst-nb==0.17.2 rst-to-myst==0.3.4 nbsphinx==0.8.8 \
-    sphinx-book-theme==0.3.2 sphinx-copybutton==0.5.2 sphinx-design==0.4.1 sphinx-prompt==1.5.0  \
-    sphinxcontrib-bibtex==2.5.0 sphinx-tabs==3.2.0 \
-    exhale==0.2.3 breathe==4.14.1 sphinx-sitemap==2.5.0 ipython==8.12.1 attrs==21.4.0
+RUN pip3 install \
+      ablog \
```

Comment:
generally - fixing the versions is better / safer - I would prefer to selectively upgrade 
previously we ran into an isssue where we had a breaking change come in because the dependencies were not pinned.

Reply:
I pinned only sphinx version for now, supposing that python pip module is smart enough to install only packages with a compatible version.
But there is no right path to do it, we may not pin version and start getting in trouble or pin it and get the issue with dropped support for it…

IMO this image is kind of side car for us, we shouldn’t worry much about it.